File: 2262-total-appeal-of-a-string/2262-total-appeal-of-a-string.py
class Solution(object):
    def appealSum(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans = 0
        n = len(s)
        last = [-1] * 26
        
        for i in range(n):
            j = last[ord(s[i]) - ord('a')]
            ans += (i - j) * (n - i)
            last[ord(s[i]) - ord('a')] = i
        
        return ans

File: 2262-total-appeal-of-a-string/2262-total-appeal-of-a-string.cpp
class Solution {
public:
    long long appealSum(string s) {
        long long ans = 0; int n = s.length();
        vector<int> last(26, -1);
        for(int i = 0; i < n; ++i) {
            int j = last[s[i] - 'a'];
            ans += (long long)(i - j) * (long long)(n - i);
            last[s[i] - 'a'] = i;
        }
        return ans;
    }
};